Project-I-Social-Card/social-card/src/components/FooterComponents/Footer.js

```diff
@@ -1,28 +1,125 @@
+        text: 4,
+    },
+    {
+        id: 4,
+        name: 'icon-image envelope',
+        textClass: 'footer-text-envelope',
+        eventFunction: envelopeClicked,
+        src: 'https://img.icons8.com/ios/30/000000/new-post.png',
+        text: null,
+    }
+];
+
+function commentClicked() {
+    const commentIcon = document.querySelector('.comment');
+    if (commentIcon.style.opacity === '0.5') {
+        commentIcon.style.opacity = '1';
+        commentIcon.style.transform = 'none';
+    } else {
+        commentIcon.style.opacity = '0.5';
+        commentIcon.style.transform = 'scaleX(-1)';
+    }
+
+    const commentIconText = document.querySelector('.footer-text-comment');
+    if (commentIconText.innerHTML === '') {
+        commentIconText.innerHTML = '1';
+    } else {
+        commentIconText.innerHTML = parseInt(commentIconText.innerHTML, 10) + 1;
+    }
+
+    if (parseInt(commentIconText.innerHTML, 10) > 10) {
+        commentIconText.style.color = 'red';
+        commentIcon.style.opacity = '1';
+    }
+}
+
+function refreshClicked() {
+    const commentIcon = document.querySelector('.refresh');
+    const commentIconText = document.querySelector('.footer-text-refresh');
+    if (commentIconText.innerHTML === '') {
+        commentIconText.innerHTML = '1';
+    } else {
+        commentIconText.innerHTML = parseInt(commentIconText.innerHTML, 10) + 1;
+    }
+
+    if (parseInt(commentIconText.innerHTML, 10) > 10) {
+        commentIconText.style.color = 'red';
+        commentIcon.style.opacity = '1';
+    }
+}
+
+function heartClicked() {
+    const commentIcon = document.querySelector('.heart');
+    const commentIconText = document.querySelector('.footer-text-heart');
+    if (commentIconText.innerHTML === '') {
+        commentIconText.innerHTML = '1';
+    } else {
+        commentIconText.innerHTML = parseInt(commentIconText.innerHTML, 10) + 1;
+    }
+
+    if (parseInt(commentIconText.innerHTML, 10) > 10) {
+        commentIconText.style.color = 'red';
+        commentIcon.style.opacity = '1';
+    }
+}
+
+function envelopeClicked() {
+    const commentIcon = document.querySelector('.envelope');
+    const commentIconText = document.querySelector('.footer-text-envelope');
+    if (commentIconText.innerHTML === '') {
+        commentIconText.innerHTML = '1';
+    } else {
+        commentIconText.innerHTML = parseInt(commentIconText.innerHTML, 10) + 1;
+    }
+
+    if (parseInt(commentIconText.innerHTML, 10) > 10) {
+        commentIconText.style.color = 'red';
+        commentIcon.style.opacity = '1';
+    }
+}
+
 function Icon (props) {
+    const { icon } = props;
     return (
-        <div className='icon-container'>
-            <img src={props.src} className='icon-image' alt='' />
-            <p className='footer-text'>{props.text}</p>
+        <div onClick={icon.eventFunction} className='icon-container'>
+            <img src={icon.src} className={icon.name} alt={icon.name} />
```

Comment:
great job destructuring your icon from props and dropping it in, you could do the same with your name and text props as well.